### ISSUE TITLE
CTD when clicking on "Switch to System Overview" button

### ISSUE THREAD

**Issue by DarioTD** [pioneerspacesim/pioneer#5364]
### Observed behaviour
Pioneer crashes when I change from "normal view" to "system overview" while hypertraveling from Sol to GJ 1116

### Steps to reproduce
-Start on Mars
-Have at least 3 hidrogen to hyperjump
-Hyperjump (illegally to save time) to GJ 1116

pioneer 20220203 (9c06c9f) Linux x86_64

#### Backtrace
```
Started hyperspacing...
libpng warning: iCCP: known incorrect sRGB profile
WARNING: texture 'icons/object_star_m.png' is not power-of-two and may not display correctly
WARNING: texture 'icons/object_planet_medium_gas_giant.png' is not power-of-two and may not display correctly
WARNING: texture 'icons/object_planet_small.png' is not power-of-two and may not display correctly
WARNING: texture 'icons/object_planet_large_gas_giant.png' is not power-of-two and may not display correctly
```
```
Thread 1 "pioneer" received signal SIGSEGV, Segmentation fault.
l_sbody_attr_physics_body (l=<optimized out>) at /home/runner/work/pioneer/pioneer/src/lua/LuaSystemBody.cpp:696
696     /home/runner/work/pioneer/pioneer/src/lua/LuaSystemBody.cpp: No such file or directory.
```
```
(gdb) bt
#0  l_sbody_attr_physics_body (l=<optimized out>) at /home/runner/work/pioneer/pioneer/src/lua/LuaSystemBody.cpp:696
#1  0x00005555558cbf62 in luaD_precall (L=L@entry=0x555566945590, func=<optimized out>, nresults=<optimized out>)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:318
#2  0x00005555558cc256 in luaD_call (L=0x555566945590, func=<optimized out>, nResults=<optimized out>, allowyield=<optimized out>)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:394
#3  0x00005555558cb824 in luaD_rawrunprotected (L=L@entry=0x555566945590, f=f@entry=0x5555558c3dd0 <f_call>, ud=ud@entry=0x7fffffffd820)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:131
#4  0x00005555558cc4d3 in luaD_pcall (L=L@entry=0x555566945590, func=func@entry=0x5555558c3dd0 <f_call>, u=u@entry=0x7fffffffd820, old_top=1136,
    ef=<optimized out>) at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:595
#5  0x00005555558c544b in lua_pcallk (L=L@entry=0x555566945590, nargs=nargs@entry=1, nresults=nresults@entry=1, errfunc=errfunc@entry=6,
    ctx=ctx@entry=0, k=k@entry=0x0) at /home/runner/work/pioneer/pioneer/contrib/lua/src/lapi.c:949
#6  0x00005555558c2e86 in pi_lua_protected_call (L=L@entry=0x555566945590, nargs=nargs@entry=1, nresults=nresults@entry=1)
    at /home/runner/work/pioneer/pioneer/src/lua/core/Sandbox.cpp:177
#7  0x00005555558865af in get_method_or_attr (l=0x555566945590) at /home/runner/work/pioneer/pioneer/src/lua/LuaMetaType.cpp:76
#8  get_method_or_attr (l=0x555566945590) at /home/runner/work/pioneer/pioneer/src/lua/LuaMetaType.cpp:63
#9  0x0000555555886cd9 in l_index (l=0x555566945590) at /home/runner/work/pioneer/pioneer/src/lua/LuaMetaType.cpp:133
#10 0x00005555558cbf62 in luaD_precall (L=L@entry=0x555566945590, func=<optimized out>, nresults=<optimized out>)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:318
#11 0x00005555558cc278 in luaD_call (L=0x555566945590, func=<optimized out>, nResults=nResults@entry=1, allowyield=<optimized out>)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:394
#12 0x00005555558d90e4 in callTM (L=L@entry=0x555566945590, p3=0x555577bfd6b0, hasres=1, p2=<optimized out>, p2=<optimized out>,
    p1=<optimized out>, p1=<optimized out>, f=<optimized out>, f=<optimized out>) at /home/runner/work/pioneer/pioneer/contrib/lua/src/lvm.c:102
#13 0x00005555558d93f3 in luaV_gettable (L=L@entry=0x555566945590, t=0x555577bfd6a0, key=0x5555675df440, val=val@entry=0x555577bfd6b0)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/lvm.c:127
#14 0x00005555558daa1e in luaV_execute (L=L@entry=0x555566945590) at /home/runner/work/pioneer/pioneer/contrib/lua/src/lvm.c:588
#15 0x00005555558cc290 in luaD_call (L=0x555566945590, func=<optimized out>, nResults=<optimized out>, allowyield=<optimized out>)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:395
#16 0x00005555558cb824 in luaD_rawrunprotected (L=L@entry=0x555566945590, f=f@entry=0x5555558c3dd0 <f_call>, ud=ud@entry=0x7fffffffdc30)
    at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:131
#17 0x00005555558cc4d3 in luaD_pcall (L=L@entry=0x555566945590, func=func@entry=0x5555558c3dd0 <f_call>, u=u@entry=0x7fffffffdc30, old_top=304,
    ef=<optimized out>) at /home/runner/work/pioneer/pioneer/contrib/lua/src/ldo.c:595
#18 0x00005555558c544b in lua_pcallk (L=L@entry=0x555566945590, nargs=nargs@entry=1, nresults=nresults@entry=1, errfunc=errfunc@entry=18,
    ctx=ctx@entry=0, k=k@entry=0x0) at /home/runner/work/pioneer/pioneer/contrib/lua/src/lapi.c:949
#19 0x00005555558c2e86 in pi_lua_protected_call (L=0x555566945590, nargs=nargs@entry=1, nresults=nresults@entry=1)
    at /home/runner/work/pioneer/pioneer/src/lua/core/Sandbox.cpp:177
#20 0x00005555556dd178 in LuaTable::Call<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double> (
    key="system-view", this=0x7fffffffdcf0) at /home/runner/work/pioneer/pioneer/src/lua/LuaPushPull.h:156
#21 PiGui::RunHandler (delta=0.016598068177700043, handler="system-view") at /home/runner/work/pioneer/pioneer/src/pigui/LuaPiGui.cpp:85
#22 0x00005555556f8c20 in PiGuiView::DrawPiGui (this=<optimized out>) at /home/runner/work/pioneer/pioneer/src/Pi.h:138
--Type <RET> for more, q to quit, c to continue without paging--c
#23 0x00005555556168ed in GameLoop::Update (this=0x5555668e81f0, deltaTime=0.0165980682) at /home/runner/work/pioneer/pioneer/src/Pi.cpp:1024
#24 0x000055555598807d in Application::Run (this=0x5555668e7f10) at /home/runner/work/pioneer/pioneer/src/core/Application.cpp:187
#25 0x00005555555e2291 in main (argc=<optimized out>, argv=<optimized out>) at /home/runner/work/pioneer/pioneer/src/Pi.h:147
```

**Comment by Keeper1st** [comment]
I think this has been fixed already by #5361 -- try compiling with that PR and see.

**Comment by DarioTD** [comment]
> I think this has been fixed already by #5361 -- try compiling with that PR and see.

It still crashes :(

### Backtrace
```
Started hyperspacing...

Thread 1 "pioneer" received signal SIGSEGV, Segmentation fault.
l_sbody_attr_physics_body (l=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaSystemBody.cpp:696
696             SystemPath gamepath = Pi::game->GetSpace()->GetStarSystem()->GetPath();
```
```
(gdb) bt
#0  l_sbody_attr_physics_body (l=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaSystemBody.cpp:696
#1  0x00005555558af69e in luaD_precall (L=L@entry=0x55555661c4e0, func=<optimized out>, nresults=<optimized out>)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:318
#2  0x00005555558af9f2 in luaD_call (L=0x55555661c4e0, func=<optimized out>, nResults=<optimized out>, allowyield=<optimized out>)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:394
#3  0x00005555558aefdc in luaD_rawrunprotected (L=L@entry=0x55555661c4e0, f=f@entry=0x5555558a6060 <f_call>, ud=ud@entry=0x7fffffffd850)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:131
#4  0x00005555558afc5f in luaD_pcall (L=L@entry=0x55555661c4e0, func=func@entry=0x5555558a6060 <f_call>, u=u@entry=0x7fffffffd850, old_top=1136,
    ef=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:595
#5  0x00005555558a8877 in lua_pcallk (L=0x55555661c4e0, nargs=<optimized out>, nresults=1, errfunc=<optimized out>, ctx=<optimized out>,
    k=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/lapi.c:949
#6  0x00005555558a5102 in pi_lua_protected_call (L=L@entry=0x55555661c4e0, nargs=nargs@entry=1, nresults=nresults@entry=1)
    at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/core/Sandbox.cpp:177
#7  0x000055555586947f in get_method_or_attr (l=0x55555661c4e0) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaMetaType.cpp:76
#8  get_method_or_attr (l=0x55555661c4e0) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaMetaType.cpp:63
#9  0x000055555586964a in l_index (l=0x55555661c4e0) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaMetaType.cpp:133
#10 0x00005555558af69e in luaD_precall (L=L@entry=0x55555661c4e0, func=<optimized out>, nresults=<optimized out>)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:318
#11 0x00005555558afa10 in luaD_call (L=0x55555661c4e0, func=<optimized out>, nResults=nResults@entry=1, allowyield=<optimized out>)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:394
#12 0x00005555558bcd2c in callTM (L=L@entry=0x55555661c4e0, p1=p1@entry=0x555564247290, p2=p2@entry=0x555556719f30, p3=p3@entry=0x5555642472a0,
    hasres=1, f=<optimized out>, f=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/lvm.c:102
#13 0x00005555558bcfba in luaV_gettable (L=L@entry=0x55555661c4e0, t=0x555564247290, key=0x555556719f30, val=val@entry=0x5555642472a0)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/lvm.c:127
#14 0x00005555558be73b in luaV_execute (L=L@entry=0x55555661c4e0) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/lvm.c:588
#15 0x00005555558afa28 in luaD_call (L=0x55555661c4e0, func=<optimized out>, nResults=<optimized out>, allowyield=<optimized out>)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:395
#16 0x00005555558aefdc in luaD_rawrunprotected (L=L@entry=0x55555661c4e0, f=f@entry=0x5555558a6060 <f_call>, ud=ud@entry=0x7fffffffdc70)
    at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:131
#17 0x00005555558afc5f in luaD_pcall (L=L@entry=0x55555661c4e0, func=func@entry=0x5555558a6060 <f_call>, u=u@entry=0x7fffffffdc70, old_top=304,
    ef=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/ldo.c:595
#18 0x00005555558a8877 in lua_pcallk (L=0x55555661c4e0, nargs=<optimized out>, nresults=1, errfunc=<optimized out>, ctx=<optimized out>,
    k=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/contrib/lua/src/lapi.c:949
#19 0x00005555558a5102 in pi_lua_protected_call (L=L@entry=0x55555661c4e0, nargs=nargs@entry=1, nresults=nresults@entry=1)
    at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/core/Sandbox.cpp:177
#20 0x00005555556beaaa in LuaTable::Call<bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double> (
    key="system-view", this=0x7fffffffdd20) at /tmp/pioneer-fix-sectormap-labels-ctd/src/lua/LuaTable.h:402
#21 PiGui::RunHandler (delta=0.016806254163384438, handler="system-view") at /tmp/pioneer-fix-sectormap-labels-ctd/src/pigui/LuaPiGui.cpp:85
#22 0x00005555556da24c in PiGuiView::DrawPiGui (this=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/src/Pi.h:138
--Type <RET> for more, q to quit, c to continue without paging--c
#23 0x00005555555fba2d in GameLoop::Update (this=0x555555b8ec10, deltaTime=0.0168062542) at /tmp/pioneer-fix-sectormap-labels-ctd/src/Pi.cpp:1024
#24 0x000055555597af07 in Application::Run (this=0x555555b8e930) at /tmp/pioneer-fix-sectormap-labels-ctd/src/core/Application.cpp:187
#25 0x00005555555c9066 in main (argc=<optimized out>, argv=<optimized out>) at /tmp/pioneer-fix-sectormap-labels-ctd/src/Pi.h:147
```

**Comment by impaktor** [comment]
Thanks for reporting, and back trace. Hope it might be useful for @Gliese852 or whom ever looks at this.

**Comment by Gliese852** [comment]
Strange that it hasn't fired before. The program requests the current star system, and it, of course, in hyperspace is equal to nullptr (at best), and then calls its method. This will be a small fix.

**Comment by Bodasey** [comment]
fixed by #5428 could not reproduce in actual master
